### ISSUE TITLE
error with tf2.4.1

### ISSUE THREAD

**Issue by abred** [Rishit-dagli/Perceiver#20]
Hello Rishit,

thank you for your Perceiver implementation!
I have two notes, I am not very familiar with tf2 though.
You define and call a tf.keras.Sequential model here https://github.com/Rishit-dagli/Perceiver/blob/4d3b9b0514da4fb623d178e3e70df1836ebad5ba/perceiver/perceiver.py#L106
For my version of tf at least this throws an error, I think it should be defined once in `__init__` and then just called in `call`.

And just above it, you compute `data` but then you don't pass it to `self.model`. Is that correct?

**Comment by abred** [comment]
thanks

1. Yes, I am using 2.4.1. I am calling it within a `@tf.function`, maybe that's the difference.
This is the error message:
```    
ValueError: Weights for model sequential_25 have not yet been created. Weights are created when the Model is first called on inputs or `build()` is called with an `input_shape`.
```

2. So if I undestand it correctly, in the current state it is not really a cross attention, as the byte array/data is not passed in?
https://github.com/Rishit-dagli/Perceiver/blob/4d3b9b0514da4fb623d178e3e70df1836ebad5ba/perceiver/attention.py#L33-L34
context is None in both cases so far

**Comment by Rishit-dagli** [comment]
1. Ah got it, fixed via #21
2. Yes that is correct

I am closing this issue as of now (after #21 ) however, feel free to reach back or reopen it.
